basenode/src/main/java/io/coti/basenode/services/HealthMetric.java

```diff
@@ -372,6 +374,21 @@ public String getDescription() {
             return "The number of direct messages received via ZeroMQ.";
         }
     },
+    ZERO_MQ_SOCKET_DISCONNECTS(ZERO_MQ_SOCKET_DISCONNECTS_LABEL, MetricClass.QUEUE_METRIC, 10, 20, false, HealthMetricOutputType.EXTERNAL) {
+        public void doSnapshot() {
+            baseDoSnapshot(this, (long) ZeroMQUtils.getSocketDisconnects(SocketType.ROUTER));

```

Comment:
consider. add a flag that says to clear the list after fetching or change the method that it clears